azure-pipelines.yml

```diff
@@ -50,19 +50,42 @@ jobs:
       azure_name: rv32imc-toolchains
 
 - job: "Toolchains_RV32IMCB"
-  displayName: "GCC and Clang/LLVM toolchains targeting RV32IMC+B (Ibex with Bit Manipulation)"
+  displayName: "Toolchains targeting Ibex with 'Balanced' Bit-manipulation"
+  # These builds are using the 'Balanced' bit-manipulation subset identified in
+  # https://github.com/lowRISC/opentitan/issues/2708, and formatted for the
+  # correct versions as implemented by clang (B 0.92).
   pool:
     vmImage: "ubuntu-16.04"
   timeoutInMinutes: 360
   steps:
   - template: "_build-deps.yml"
 
   - bash: |
+      # HACK: GCC does not allow you to compile with just a subset of the B
+      # extension using `-march=`, but it does assemble separately.
+      #
+      # So, we pass a more acceptable `-march=` to the assembler with `-Wa`.
+      #
+      # However, gcc will emit an assembly directive describing the arch it
+      # wanted when it compiled, which the assembler will use in spite of the
+      # `-march=` flag. So, we have to turn this off, using
+      # `-mno-riscv-attribute`.
+      #
+      # The end result is: The compiler will not introduce B extension
+      # instructions when compiling, but the assembler will understand them, so
+      # they can be used in inline assembly.
+      #
+      # If you update the patches in `patches/lowrisc-toolchain-gcc-rv32imcb`,
+      # you will need to revalidate that this behaviour works. A good testcase
+      # is a file with a `pcnt` instruction in inline assembly.
+      #
       ./build-gcc-with-args.sh \
         "lowrisc-toolchain-gcc-rv32imcb" \
         "riscv32-unknown-elf" \
         "/tools/riscv" \
-        "-march=rv32imcb" "-mabi=ilp32" "-mcmodel=medany"
+        "-march=rv32imc" "-mabi=ilp32" "-mcmodel=medany" \
+        "-Wa,-march=rv32imczbb0p92_zbs0p92_zbf0p92_zbt0p92" \
+        "-mno-riscv-attribute"
```

Comment:
Hey @lenary ,
sorry for the long delay. I have tested this using a simple application containing an inline assembly Bitmanip instruction and with this toolchain: `lowrisc-toolchain-gcc-rv32imcb-20200626-1`.

It works but the subgroup flags need to be provided in alphabetical order. Meaning this needs to be changed to:
```
        "-Wa,-march=rv32imczbb0p92_zbf0p92_zbs0p92_zbt0p92" \
        "-mno-riscv-attribute"
```

Reply:
When doing this check, I was providing `-march=rv32imc`.